custom-login/index.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/******************************************************************************
+ * Copyright 2018 Okta, Inc.                                                  *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License");            *
+ * you may not use this file except in compliance with the License.           *
+ * You may obtain a copy of the License at                                    *
+ *                                                                            *
+ *      http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ ******************************************************************************/
+
+require_once __DIR__ . '/vendor/autoload.php';
+
+$loader = new Twig_Loader_Filesystem(__DIR__ . '/views');
+$twig = new Twig_Environment($loader);
+
+$dotenv = new Dotenv\Dotenv(__DIR__);
+$dotenv->overload();
+
+$state = 'applicationState';
+
+$dispatcher = FastRoute\simpleDispatcher(function(FastRoute\RouteCollector $r) use ($twig, $state) {
+    $r->get('/', function() use ($twig) {
+        echo $twig->render('index.twig', [
+            'authenticated' => isAuthenticated(),
+            'profile' => getProfile()
+        ]);
+    });
+
+    $r->get('/login', function() use ($twig, $state) {
+        $baseUrlParts = parse_url(getenv('ISSUER'));
+        $_ENV['BASE_URL'] = $baseUrlParts['scheme']."://".$baseUrlParts['host'];
+
+        echo $twig->render('login.twig',[
+            'env' => $_ENV,
+            'state' => $state
+        ]);
+    });
+
+    $r->get('/authorization-code/callback', function() use ($state) {
+        if(array_key_exists('state', $_REQUEST) && $_REQUEST['state'] !== $state) {
+            throw new \Exception('State does not match.');
+        }
+
+        if(array_key_exists('code', $_REQUEST)) {
+            $exchange = exchangeCode($_REQUEST['code']);
+            if(!isset($exchange->access_token)) {
+                die('Could not exchange code for an access token');
+            }
+
+            if(verifyJwt($exchange->access_token) == false) {
+                die('Verification of JWT failed');
+            }
+
+
+            setcookie("access_token","$exchange->access_token",time()+$exchange->expires_in,"/",false);
```

Comment:
The cookie is what I am using to store the access_token and figure out if the user is authenticated or not.   This is the way it was in the original samples, and the way it was done and recommended by both SP and Myself at every talk about auth I give. The only two items as part of this cookie that I am missing are:

1: make it secure(https) only, which I cant do because it requires a lot of setup to make localhost https capable
2: make it serve over http only (no clientside js), which I just missed here